pramen-py/src/pramen_py/utils/__init__.py

```diff
@@ -148,7 +148,11 @@ def get_or_create_spark_session(
             "JAVA_HOME is set to " + env.str("PRAMENPY_SPARK_JAVA_HOME")
         )
 
-    maybe_active_session = SparkSession.getActiveSession()  # type: ignore
+    if callable(getattr(SparkSession, "getActiveSession", None)):
+        maybe_active_session = SparkSession.getActiveSession()
+    else:
+        maybe_active_session = SparkSession.getOrCreate()
```

Comment:
If `getActiveSession` attribute is absent, you should not create a Spark session. This is beacuse later we create it adding extra config options